### PR TITLE
Some griddle fixes

### PR DESCRIPTION
**Pull request by ghost**
:cl: Qustinnus
fix: The griddle no longer leaves null objects in the griddle, causing it to seem full
fix: Bonfires can now be used as ghetto griddles just like with normal grills
fix: You can now deconstruct griddles
/:cl:

fixes some parts of #55543 (comeon use separate issue reports)

### CODE REVIEW

**Comment by Owai-Seek** [comment]
Suppose I could always close the issue (if it doesn't auto-close) and re-make it if the null objects thing doesn't fix it.

**Comment by Owai-Seek** [comment]
Posted what I (think) is the underlying issue on the issue report. Heard two people mention bacon, and investigated it.

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 1]
```diff
@@ -27,6 +27,11 @@
 	grill_loop = new(list(src), FALSE)
 	variant = rand(1,3)
 
+/obj/machinery/griddle/crowbar_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(default_deconstruction_crowbar(I, TRUE))
```
Comment (Jared-Fogle) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:32]: Can you make this used named parameters for the TRUE?

File: code/modules/food_and_drinks/kitchen_machinery/griddle.dm [hunk 2]
```diff
@@ -61,19 +66,26 @@
...
+	UnregisterSignal(I, list(COMSIG_GRILL_COMPLETED, COMSIG_MOVABLE_MOVED, COMSIG_PARENT_QDELETING))
 	update_grill_audio()
 
+/obj/machinery/griddle/proc/ItemMoved(obj/item/I, atom/OldLoc, Dir, Forced)
+	SIGNAL_HANDLER
+	ItemRemovedFromGrill(I)
+
+/obj/machinery/griddle/proc/ItemDeleted(datum/source)
+	SIGNAL_HANDLER
+	ItemRemovedFromGrill(source)
```
Comment (Jared-Fogle) [inline code/modules/food_and_drinks/kitchen_machinery/griddle.dm:85]: Why not `RegisterSignal(item_to_grill, COMSIG_PARENT_QDELETING, .proc/ItemRemovedFromGrill)`?

**Review by Mothblocks — APPROVED** [review]
why do they call it oven when you of in the cold food of out hot eat the food

**Pull request merged**